CMakeLists.txt

```diff
@@ -20,15 +23,6 @@ list(APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake)
 
 option(BUILD_SHARED_LIBS "Build libraries as shared as opposed to static" ON)
 
-# Enable C++14
-set(CMAKE_CXX_EXTENSIONS OFF)
-set(CMAKE_CXX_STANDARD 14)
-set(CMAKE_CXX_STANDARD_REQUIRED 11)
```

Comment:
Why did you remove this? Btw, it should be ``set(CMAKE_CXX_STANDARD_REQUIRED ON)``

Reply:
`set(CMAKE_CXX_EXTENSIONS OFF)` should not be removed (it's my fault).

While
```cmake
set(CMAKE_CXX_STANDARD 14)
set(CMAKE_CXX_STANDARD_REQUIRED 11)
```
can be removed if I add `target_compile_features(${LIBRARY_TARGET_NAME} PUBLIC cxx_std_14)`. If I'm not mistaken adding this kind of commands may lead to problem when the library is compiled within other projects (@traversaro  may confirm/deny)

Reply:
In 24b77cac3a19e43ae2bbf663b2ee4a450d5130ca i restored `set(CMAKE_CXX_EXTENSIONS OFF)` and  in 69a410aeab45b280500b5f13941f38bab6d48c05 i required `c++14`  for compiling the library